File: models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: [true, 'Category name is required'],
      trim: true,
      unique: true,
      maxlength: [50, 'Category name cannot exceed 50 characters'],
    },
    color: {
      type: String,
      default: '#6366f1',
      match: [/^#([0-9A-F]{3}){1,2}$/i, 'Invalid hex color'],
    },
    icon: {
      type: String,
      default: '📁',
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model('Category', categorySchema);

File: models/category.py
"""
models/category.py
──────────────────
Beanie document → maps to the 'categories' collection in MongoDB.
Pydantic validates every field automatically.
"""

from beanie import Document
from pydantic import Field


class Category(Document):
    name: str = Field(..., max_length=50)
    color: str = Field(default="#6366f1")   # hex colour shown in the UI
    icon: str = Field(default="📁")

    class Settings:
        name = "categories"   # MongoDB collection name
